.gitlab-ci.yml

```diff
@@ -52,73 +53,131 @@ variables:
+    - for crate in "${wasm_crates[@]}"; do
+        cargo clippy --verbose --all-features -- -D warnings;
+      done
+  allow_failure:                   true
 
-cargo-build-nf:
+clippy-wasm:
   stage:                           check-workspace
   <<:                              *docker-env
   script:
-  - cargo build --verbose --all --no-default-features --release --target=wasm32-unknown-unknown
+    - wasm_crates=("core" "alloc" "utils" "lang2" "lang2/macro")
+    - for crate in "${wasm_crates[@]}"; do
+        cargo clippy --verbose --manifest-path ${crate}/Cargo.toml --no-default-features -- -D warnings;
+      done
+  allow_failure:                   true
+
+fmt:
+  stage:                           check-workspace
+  <<:                              *docker-env
+  script:
+    - all_crates=("core" "alloc" "utils" "lang2" "lang2/macro" "cli")
+    - for crate in "${wasm_crates[@]}"; do
+        cargo fmt --verbose -- --check;
+      done
+  allow_failure:                   true
+
 
 #### stage:                        examples
 
-examples-wasm:
+examples-test:
   stage:                           examples
   <<:                              *docker-env
   script:
-    - for example in $(ls examples/lang); do
-        cargo build --release --no-default-features --target=wasm32-unknown-unknown --verbose --manifest-path examples/lang/$example/Cargo.toml;
+    - for DIR in examples/lang2/*; do 
+        cargo test --verbose --manifest-path ${DIR}/Cargo.toml;
       done
-    - sccache -s
 
-examples-test:
+examples-fmt:
   stage:                           examples
   <<:                              *docker-env
   script:
-    - for example in $(ls examples/lang); do
-        cargo test --verbose --manifest-path examples/lang/$example/Cargo.toml;
+    - for DIR in examples/lang2/*; do 
+        cargo fmt --verbose --manifest-path ${DIR}/Cargo.toml -- --check;
       done
-    - sccache -s
+  allow_failure:                   true
 
-examples-abi:
+examples-clippy-all-features:
   stage:                           examples
   <<:                              *docker-env
   script:
-    - for example in $(ls examples/lang); do
-        cargo run --package abi-gen --manifest-path examples/lang/$example/Cargo.toml;
+    - for DIR in examples/lang2/*; do 
+        cargo clippy --verbose --manifest-path ${DIR}/Cargo.toml --all-features -- -D warnings;
+      done
+  allow_failure:                   true
+
+examples-clippy-no-default-features:
+  stage:                           examples
+  <<:                              *docker-env
+  script:
+    - for DIR in examples/lang2/*; do 
+        cargo clippy --verbose --manifest-path ${DIR}/Cargo.toml --no-default-features -- -D warnings;
+      done
+  allow_failure:                   true
+
+examples-contract-build:
+  stage:                           examples
+  <<:                              *docker-env
+  script:
+    - for DIR in examples/lang2/*; do 
+        cd $DIR && pwd && cargo contract build; 
```

Comment:
I'd go with a
```
pushd $example && cargo contract build && popd;
```

Reply:
That's the same

Reply:
It is a more robust solution in general.